libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,435 @@
+- `titles`: Typography h3, h4 elements
+- `body`: Define typography for base body text.
+- `buttons`: Typography for buttons and anchors
+
+> **_NOTE:_**  Don't forget to check if you [corrected configure](#importing-fonts) or import font family's
+
+```scss
+$custom-theme: (
+ general: (
+  density: 0,
+  colors: (
+   primary: #2581C4,
+   accent: #EA518D,
+   font: #323232,
+   disabled: (
+    foreground: rgb(0 0 0 / .38),
+    background: rgb(0 0 0 / .12)
+   )
+  ),
+  fonts: (
+   hero-titles: (
+    family: 'Helvetica',
+    size: 40px
+   ),
+   titles: (
+    family: 'Roboto',
+    size: 25px
+   ),
+   body: (
+    family: 'Helvetica',
+    size: 20px
+   ),
+   buttons: (
+    family: 'Roboto',
+    size: 10px
+   )
+  ),
+ )
+);
+```
+
+To apply it on whole body of the app, add `.mat-body` class to the body element in index.html file in app root
+
+```html
+<!-- Applying mat-body class to the root component in index.html -->
+<body class="mat-body">
+<app-root></app-root>
+</body>
+```
+
+Since we're overriding Angular Material, it is important to add CSS classes for mixin that emits styles for native header elements scoped within the .mat-typography CSS class
+
+| CSS class      | Example                                       |
+| -------------- | --------------------------------------------- |
+| .mat-headline-1   | `<h1 class="mat-headline-1">{{title}}</h1>`|
+| .mat-headline-2   | `<h2 class="mat-headline-2">{{title}}</h2>`|
+
+#### Spacing
+
+The submap providing the necessary spacing information for styling the **entry form and search form** **button** and **form field**
+
+```scss
+$custom-theme: (
+ general: (
+  spacing: (
+   default: 15px
+  ),
+ )
+);
+```
+
+#### Buttons
+
+Additional customization  for buttons:
+
+- `icon-size` - defines the default icon size for buttons.
+
+Code Example of general submap configuration:
+
+```scss
+$custom-theme: (
+ general: (
+  buttons: (
+   icon-size: 48px
+  )
+ )
+)
+```
+
+### Tables Configuration
+
+The `tables` section in the `$custom-theme` configuration handles how table components are styled and themed.
+
+#### Cells
+
+Contains properties related to individual cells in a table:
+
+- `edge-gap` specifies the gap between cell edges.
+- `padding` allows customization of cell padding.
+
+#### Rows
+
+Handles the appearance of table rows:
+
+- `selected-color` sets the background color for selected rows.
+- `disabled-color` defines the background color for disabled rows.
+- `odd-even-row` determines whether odd or even rows are styled differently.
+- `odd-even-background` sets the background color for odd or even rows.
+
+#### Contents
+
+Manages the styling of table content:
+
+- `no-result` contains properties for the appearance of a message displayed when there are no search results:
+- `font-size` specifies the font size of the message.
+- `font-weight` controls the font weight of the message.
+
+```scss
+$custom-theme: (
+ tables: (
+  cells: (
+   edge-gap: 4px,
```

Comment:
We should either explain by words or add the most complex examples for setting up everything; i.e. edge-gap can also be 4px 8px 3px 5px
Row can be either odd or even, nothing else.